src/openst.js

```diff
@@ -68,9 +70,60 @@ class OpenST {
       createAndAddModules: createAndAddModules.address,
       tokenRules: tokenRulesAddress,
     };
+
     Object.assign(this.chainConfig.openst, setupData);
+
+    this.chainConfig.utilityBrandedTokens[ubtConfig.index].openst = setupData;
+
     logger.info('Completed Setup of OpenST');
   }
+
+  getUtilityBrandedTokenConfig(auxiliaryEIP20Token) {
+    for (let i = 0; i < this.chainConfig.utilityBrandedTokens.length; i++) {
+      const ut = this.chainConfig.utilityBrandedTokens[i];
+      if (ut.address === auxiliaryEIP20Token) {
+        return { ubt: ut, index: i };
+      }
+    }
+    return 0;
+  }
+
+  async deployPricerRule(
+    auxiliaryEIP20Token,
+    baseCurrencyCode,
+    conversionRate,
+    conversionRateDecimals,
+    requiredPriceOracleDecimals,
+  ) {
+    logger.info('Deployment of PricerRule');
+    const ubtConfig = this.getUtilityBrandedTokenConfig(auxiliaryEIP20Token);
+
+    const { PricerRule } = Package.ContractInteract;
+
+    const PricerRuleTxOptions = this.auxiliary.txOptions;
+    const pricerRule = await PricerRule.deploy(
+      this.auxiliary.web3,
+      ubtConfig.ubt.organizationAddress,
+      ubtConfig.ubt.address,
+      ubtConfig.ubt.openst.tokenRules,
+      baseCurrencyCode,
+      conversionRate,
+      conversionRateDecimals,
+      requiredPriceOracleDecimals,
+      PricerRuleTxOptions,
+    );
+
+    const pricerOracleData = {
+      address: pricerRule.address,
+      baseCurrencyCode,
+      conversionRate,
+      conversionRateDecimals,
+      requiredPriceOracleDecimals,
+    };
+
+    this.chainConfig.utilityBrandedTokens[ubtConfig.index].openst.pricerOracle = pricerOracleData;
```

Comment:
I don't think we need to write this data in config. We just need to write pricerRule contract address.
If we have pricerRule contract address we can fetch it from pricerRule contract address.

Also why it's being written under pricerOracle 😑 
```
.openst.pricerOracle
```